### PR TITLE
Update for Conan build

### PR DESCRIPTION
**Pull request by pato-melowntech**

### CODE REVIEW

File: buildsys.cmake [hunk 1]
```diff
@@ -42,10 +42,10 @@ macro(cpp_msvc_overrides)
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/Ot>)
 
   # disable security checks (stack buffer overrun prevention)
-  add_compile_options($<$<AND:$<COMPILE_LANGUAGE:C,CXX>,$<CONFIG:RELEASE>>:/GS->)
-  add_compile_options($<$<AND:$<COMPILE_LANGUAGE:C,CXX>,$<CONFIG:RELEASE>>:/O2>)
-  add_compile_options($<$<AND:$<COMPILE_LANGUAGE:C,CXX>,$<CONFIG:RELWITHDEBINFO>>:/GS->)
-  add_compile_options($<$<AND:$<COMPILE_LANGUAGE:C,CXX>,$<CONFIG:RELWITHDEBINFO>>:/O2>)
+  add_compile_options($<$<AND:$<COMPILE_LANGUAGE:C,CXX>,$<CONFIG:Release>>:/GS->)
```
Comment (zap150) [inline buildsys.cmake:45]: Don't we call `string(TOUPPER "${CMAKE_BUILD_TYPE}" BT)` before this?

File: buildsys.cmake [hunk 2]
```diff
@@ -194,21 +215,16 @@ endmacro()
 
 # enable C++17
 macro(enable_cpp17)
+  set(CMAKE_C_STANDARD 11)
```
Comment (zap150) [inline buildsys.cmake:218]: Do we want to mix C and C++ standards?

Comment (vaclavblazek) [reply buildsys.cmake:218]: C11 is the last C standard, it has nothing to do with C++ standard versioning.

Comment (vaclavblazek) [reply buildsys.cmake:218]: I stand corrected, _they_ released C17 and even C23.

File: buildsys.cmake (hunk 1 above)
Comment (aroidzap) [reply buildsys.cmake:45]: This does not change the CMAKE_BUILD_TYPE variable, and:

> CMAKE_BUILD_TYPE Specifies the build type on single-configuration generators.
https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html

Comment (pato-melowntech) [reply buildsys.cmake:45]: This does not change the CMAKE_BUILD_TYPE variable, and:

CMAKE_BUILD_TYPE Specifies the build type on single-configuration generators.
https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html

**Review by zap150 — APPROVED** [review]

**Pull request merged**